C12 installfest (#569)
Ada-Developers-Academy/textbook-curriculum: 00-programming-fundamentals/installfest.md

For example, a guide may say to do `$ pwd` to print the path of the current directory. What you literally do is type `pwd` and enter into the Terminal.

## Xcode Command-Line Tools

Xcode is an **integrated development environment** (often called an **IDE**) for creating apps for Mac, iPhone, and iPad. We don't need it in it's entirety, we're just going to download one piece of the whole. _Command Line Tools_ is a component of Xcode that provides a _compiler_ for OS X.

> **You should have installed Xcode Command-Line tools in JumpStart.**

Reviewer: ```suggestion
**You should have installed Xcode Command-Line tools in [JumpStart](https://github.com/Ada-Developers-Academy/jump-start/tree/master/preparing-to-code/environment-setup#installfest).**
```

If the command does not show a directory path and instead something like `xcode-select: error: unable to get active developer directory, use xcode-select --switch to set one (or see man xcode-select)` you can install it with `$ xcode-select --install` select install and then grab coffee because it will take a bit of time.

## Homebrew

> [Homebrew](http://brew.sh/) is a **package manager** that simplifies installing software on macOS. Package managers are very common, and each language/framework/stack tends to have their own or preferred tool.  Homebrew should also have been installed in Jumpstart.

Reviewer: ```suggestion
[Homebrew](http://brew.sh/) is a [**package manager**](https://en.wikipedia.org/wiki/Package_manager) that simplifies installing software on macOS. Package managers are very common, and each language/framework/stack tends to have their own or preferred tool.  Homebrew should also have been installed in [Jumpstart](https://github.com/Ada-Developers-Academy/jump-start/tree/master/preparing-to-code/environment-setup#installfest).
```

To install, copypasta this into your terminal:

> `$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`

Reviewer: Maybe: 

```bash
$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```

You may need to enter your computer password to complete the installation.

### Trust but Verify Brew

> Run `$ brew doctor`. Brew is super great at telling you what else it may need. Sometimes there are file permissions to fix. Sometimes there's a warning that can be ignored. If there's anything ever wrong with Brew, `brew doctor` will usually be able to tell you what it is and how to fix it.

Reviewer: I tend to think that `$`s at the beginnings of lines are confusing inline.  (Though we do this all over the place here so I get if you want to leave it.)

```suggestion
Run `brew doctor`. Brew is super great at telling you what else it may need. Sometimes there are file permissions to fix. Sometimes there's a warning that can be ignored. If there's anything ever wrong with Brew, `brew doctor` will usually be able to tell you what it is and how to fix it.
```

## RVM

[RVM](http://rvm.io) (Ruby Version Manager) is a command-line tool which allows you to easily install, manage, and work with multiple ruby environments from interpreters to sets of gems. In essence, RVM is a package manager specifically for working with Ruby code.

> [**You should have installed RVM in JumpStart.**](https://github.com/Ada-Developers-Academy/jump-start/blob/master/lessons/07-environment-setup/notes/installfest.md)

Reviewer: It looks like this file moved: 

```suggestion
[**You should have installed RVM in JumpStart.**](https://github.com/Ada-Developers-Academy/jump-start/tree/master/preparing-to-code/environment-setup#installfest)
```

### Trust but Verify RVM

- `$ rvm version` should tell you something like _rvm 1.26.11 (latest)..._

## Upgrade and Use Ruby version 2.5.5

We will use Ruby version 2.5.5 for the Ada curriculum _(last updated: May 2019)_.

- Use rvm to install the Ruby version 2.5.5 onto your computer: `$ rvm install 2.5.5`
- Verify that this version of Ruby was installed and that `rvm` recognizes it with `$ rvm list`
- Use rvm to state that you want to use version 2.5.5 and you want to set it as the default version with `$ rvm use 2.5.5 --default`

> **Note:** If you do not set this as the default version now, you will have to run `$ rvm use 2.5.5` every project! So go ahead and set it as default right now. ;)

Reviewer: "Now" implies that this is their one-and-only chance to set the default version.

```suggestion
**Note:** If you do not set this as the default version, you will have to run `$ rvm use 2.5.5` every project! So go ahead and set it as default right now. ;)
```
